### DISCUSSION TITLE
Stm32f401ccu6 compile successfully, but displays ILI9486 and ILI9342C have white color

### DISCUSSION THREAD

**Discussion by brightproject** [stm32duino/Arduino_Core_STM32#2035] [Libraries]
When I used the blue pill with [Arduino_STM32](https://github.com/rogerclarkmelbourne/Arduino_STM32) and the MCUFRIEND_kbv library, the display worked fine on the 8-bit 8080 interface
![2023-03-20 21-57-29](https://github.com/stm32duino/Arduino_Core_STM32/assets/1788098/b30cef8d-ad58-4078-a045-e843f7946cdc)
```
    //LCD pins  |D7 |D6 |D5 |D4 |D3 |D2 |D1 |D0 | |RD |WR |RS |CS |RST| |SD_SS|SD_DI|SD_DO|SD_SCK| !!!WORKED!!!
    //STM32 pin |PA7|PA6|PA5|PA4|PA3|PA2|PA1|PA0| |PB0|PB6|PB7|PB8|PB9| |PA15 |PB5  |PB4  |PB3   | **ALT-SPI1**
```
When I switched to the kernel [Arduino_Core_STM32](https://github.com/stm32duino/Arduino_Core_STM32) and began to use the [TFT_eSPI](https://github.com/Bodmer/TFT_eSPI), two of my displays do not display anything, except for a white display.
Display 1
![photo1685817698](https://github.com/stm32duino/Arduino_Core_STM32/assets/1788098/24acdf5f-0f2b-439b-99a1-3d8ddb7e3ce1)
Display 2
![photo1685696835 (1)](https://github.com/stm32duino/Arduino_Core_STM32/assets/1788098/8fc6d274-34a5-4582-8e0c-4c4940843dd0)
At the same time, the code compiles without errors, is loaded via STLink into a black pill and outputs information via UART, but images, incl. no data is sent to the pins D0-D7.
I have suspicions about fake black pills and a problem with compiler or stm32 core.
P.S. Arduino IDE 1.8.19

**Comment by fpistm** [comment]
Hi @brightproject 
unfortunately, it is probably linked to the library as I've explained in the other discussion, it seems port for each board/MCU are very specific and linked to the core used. I could not investigate on this. Your best chance is to get support from the library maintainer

**Comment by bittmanv** [comment]
Hi, I have particular experience with MCUFriend library, your issuse is tricky, on picture with Color display, you are using BluePill ( STM32F103). This is supported by MCU Friend and with 8080 interface works, but library is using direct IO Register access IO pins, which will be on black pill F401 different (I read it on some discussion ).

**Reply by brightproject** [reply]
> you are using BluePill


Yes, on photo BluePill, but this simpli old photo.
![work_eTFT_sPI](https://github.com/stm32duino/Arduino_Core_STM32/assets/1788098/6815f1ab-831d-4393-b706-f856b6fc33d8)
I have problem with STM32F401CCU6 + ST core for ILI9342C driver only.

> MCUFriend library

This library worked for stm32f401c8t6 on Rojer Clark only.
About access IO pins - yes, for BlackPill different from BluePill.
I couldn't get the library MCUFriend to work on the new kernel from STMicroelectronics...somewhere I also came across information on the forum, but I couldn't implement compatibility.
Have you worked with compiling code for STM32 using arduino cli or make?

**Reply by bittmanv** [reply]
Sorry for slow response.... 
If you have issue between Roger and STM Core (and you have logical analyser ), try to monitor RST and control pins. I had similar issue with my display (8080 interface based ) and issue was wrong/missing reset pin handling between cores. 
Resolved as github issue by MCUFriend girhub.
